routes/api/market.js

```diff
@@ -0,0 +1,85 @@
+const router = require('express').Router();
+const { poster, posterBid } = require('../../controllers/market/poster');
+const { bidUpdate } = require('../../controllers/market/updater');
+const { Market,Bid,Op } = require('../../databases/db');
+
+router.get('/', async(req,res)=>{
+    let {page = 0, size = 10} = req.query;
+    page = parseInt(page);
+    size = parseInt(size);
+    let options = {
+        offset: page * size,
+        limit: size,
+        order: [
+            ['createdAt', 'DESC']
+        ],
+        where: {
+            isFinished: false
+        }
+    };
+
+    const marketActives= await Market.findAll(options);
+    if (marketActives==''){
+        res.json({error: 'No hay ninguna subasta en linea'});
+    }else{
+        res.status(200).json({
+            success: true,
+            items : marketActives
+        });
+    }
+});
+
+//endpoint para buscar subastas por su id
+router.get('/:marketId', async (req,res)=>{
+    const market= await Market.findOne({where:{id: req.params.marketId}});
+     if(!market){
+        return res.json({
+          success: false,
+          error:'No existe esta subasta'
+        });
+    }
+    
+    //Para saber la puja más alta de la subasta  
+    const marketBid = await Bid.findAll({
+        raw:true,
+        where:{marketId:req.params.marketId}
+    });
+    const user = JSON.parse(JSON.stringify(marketBid));
+    const items = [];
+    for (let i = 0; i < user.length; i++) {
+        items.push(user[i]);
+    }
+    const max = Math.max.apply(Math, items.map(function(o) { return o.value; }))
+    const winner = items.find(item => item.value === max);
```

Comment:
Encalsulalo en una funcion aparte


```
function async getWinnerBid(marketId) {
    const marketBid = await Bid.findAll({
        raw:true,
        where:{ marketId }
    });
    const user = JSON.parse(JSON.stringify(marketBid));
    const items = [];
    for (let i = 0; i < user.length; i++) {
        items.push(user[i]);
    }
    const max = Math.max.apply(Math, items.map(function(o) { return o.value; }))
    const winner = items.find(item => item.value === max);
    return winner;
}
```

```suggestion
    const marketBid = await getWinnerBid(req.params.marketId);
```

Lejos de eso... mrc el obtener TODAS las pujas de unas ubasta me parece un poquito poco optimo para esto, creo que deberia haber una forma de ordenarlas descendentemente o algo asi y obtenerla de una, pero bueno..... igual tampoco es tan terrible supongo no existiara una sola subasta con 3510501501 pujas, solo digo que podria mejorar

Reply:
A ver creo que se podria mejorar pero podriamos dejarlo mientras asi y corregirlo en otro PR mientras middle trabaja con esto funcional eso si dejame ver si eso lo puedo encapsular